src/main/java/com/adyen/service/TerminalCloudAPI.java

```diff
@@ -83,4 +82,22 @@ public TerminalAPIResponse sync(TerminalAPIRequest terminalAPIRequest) throws IO
         }.getType());
         return terminalAPIResponse;
     }
+
+    /**
+     * Checks if SaleToAcquirerData is set. If it is not creates an encoded base 64 string with application info.
+     *
+     * @param terminalAPIRequest TerminalAPIRequest
+     * @return TerminalAPIRequest
+     */
+    private static TerminalAPIRequest getApplicationInfo(TerminalAPIRequest terminalAPIRequest) {
+        if (terminalAPIRequest != null
+                && terminalAPIRequest.getSaleToPOIRequest() != null
+                && terminalAPIRequest.getSaleToPOIRequest().getPaymentRequest() != null
+                && terminalAPIRequest.getSaleToPOIRequest().getPaymentRequest().getSaleData() != null
+                && terminalAPIRequest.getSaleToPOIRequest().getPaymentRequest().getSaleData().getSaleToAcquirerData() == null) {
```

Comment:
This validation makes more sense in the `TerminalAPIRequest` object as a `isSaleToAcquirerDataSet` method